### PR TITLE
Regenerate ONNX tests using newer torch-mlir.

### PR DESCRIPTION
**Pull request by ScottTodd**
Generated with
* https://github.com/llvm/torch-mlir/commit/e48fe4588631e7a37a2899f9d4cd5c4cbc967481 (which IREE is updating to at https://github.com/openxla/iree/pull/16607)
  * Included: https://github.com/llvm/torch-mlir/commit/e48fe4588631e7a37a2899f9d4cd5c4cbc967481 (support regions, strings, lists, optional, etc. in the importer)
  * Included: https://github.com/llvm/torch-mlir/commit/13553d49c9488e09fec6ba790fb095eea66c48ea (create `none` for missing operands - these could be cleaned up with DCE/CSE/etc.)
* IREE compiler version 20240301.817 @ https://github.com/openxla/iree/commit/e99179861c9e2af4796f51e09de271c885e88836

(Would be nice to put that version information in the repo / test case files somewhere...)

Unfortunately, none of the newly importable tests compile:
* Before: 238 passed, 809 xfailed
* After: 238 passed, 995 xfailed

### CODE REVIEW

File: iree_tests/onnx/node/import_failures.txt
Comment (ScottTodd) [inline iree_tests/onnx/node/import_failures.txt:1]: Note that we're down to 48 import failures now, compared to just 2 upstream in torch-mlir. The difference there comes from conftest.py not handling all input and output protobufs from the ONNX test suite (e.g. sequence types, optional types). To support those (if we care), we'd need to write the conversion code from ONNX protos to .npy or .bin files. ONNX has helper functions here: https://github.com/onnx/onnx/blob/b48d763a2d22574540f51b6bc5ce1a537fb65024/onnx/backend/test/runner/__init__.py#L510-L523 but I wasn't quite sure how to get those saved to .npy files.

**Comment by ScottTodd** [comment]
Ping? Want to review? (I can also just merge myself with no review :P)

**Comment by kumardeepakamd** [comment]
[AMD Official Use Only - General]

Sorry, could not get to it sooner. Will review by 11 am PST.

**Review by kumardeepakamd — APPROVED** [review]
Scanned the files changed. Looks good.

**Pull request merged**